remoting/src/main/java/org/apache/rocketmq/remoting/protocol/heartbeat/HeartbeatData.java

```diff
@@ -53,9 +56,38 @@ public void setConsumerDataSet(Set<ConsumerData> consumerDataSet) {
         this.consumerDataSet = consumerDataSet;
     }
 
+    public int getHeartbeatFingerprint() {
+        return heartbeatFingerprint;
+    }
+
+    public void setHeartbeatFingerprint(int heartbeatFingerprint) {
+        this.heartbeatFingerprint = heartbeatFingerprint;
+    }
+
+    public boolean isWithoutSub() {
+        return isWithoutSub;
+    }
+
+    public void setWithoutSub(boolean withoutSub) {
+        isWithoutSub = withoutSub;
+    }
+
     @Override
     public String toString() {
         return "HeartbeatData [clientID=" + clientID + ", producerDataSet=" + producerDataSet
             + ", consumerDataSet=" + consumerDataSet + "]";
     }
+
+    public int computeHeartbeatFingerprint() {
+        HeartbeatData heartbeatDataCopy = JSON.parseObject(JSON.toJSONString(this), HeartbeatData.class);
```

Comment:
the performance issues in the code, especially when there is a large MQ cluster and many subscribed topics

Reply:
![image](https://github.com/apache/rocketmq/assets/13689954/adeca391-b6e9-4699-8203-dc4ec70cc4ec)
Thank for your review at first，when a cid subscribe 500 topics, the computeHeartbeatFingerprint cost 70ms(In a 30s heartbeat cycle, it's not very much, but our broker can free up a lot of computation and reduce network bandwidth pressure).